### DISCUSSION TITLE
Hotspot Tooltip on Mouse-over? [event handlers]

### DISCUSSION THREAD

**Discussion by SolidModeler** [cnr-isti-vclab/3DHOP#45] [Q&A] [answered]
3DHOP is amazing - excellent work! 
I'd like to present users with a nice tooltip when they hover over hotspots.

The 3DHOP Hotspot tutorial works great as described using the JavaScript Alert function fired by the `_onEnterSpot` event. So I tried substituting the `_onLeaveSpot` with the `_onEnterSpot` mouse-over event.  But the Alert was no longer invoked. 

The respective parts of the code are:
`presenter._onEnterSpot = onEnterSpot;
`
and
```
function onEnterSpot(id) {
  switch(id) {
     case 'Spot1'   : alert("Thing 1"); break;
     case 'Spot2' : alert("Thing 2"); break;
  }
}
```
Is it obvious what I'm doing wrong? 
I figured I'd better ask before attempting to build a handler for the tooltip itself.

**Comment by potenziani** [comment, marked as answer]
Dear @SolidModeler,

sorry for the huge delay in the reply! You probably already solved the problem by yourself, however, since this answer might be useful to other users, here is a tip for using the `onEnterSpot` feature in the right way!

Your code seems to be ok, the only thing that could be wrong is the unique ID of the hotspots raising the events (`Spot1` and `Spot2` in this case). As you know the correspondence between the hotspots/instances unique IDs used in the `setScene` definition and those used in the event handlers provided by 3DHOP is essential to get proper functioning.

Let's see an example! If you try to edit the Hotspots and Events how-to (3.2), the respective parts of the code should be:

`presenter._onEnterSpot = onEnterSpot; `

and
```
function onEnterSpot(id) {
  switch(id) {
     case 'Wing'   : alert("Thing 1"); break;
     case 'Sphere' : alert("Thing 2"); break;
  }
}
```

This should work (if not, let us know), giving you the possibility to build a nice tooltip without rewriting a tooltip handler!
